docs/writing-rules/pattern-syntax.mdx

```diff
@@ -515,7 +515,37 @@ When referencing an ellipsis metavariable in a rule message or [metavariable-pat
 - message: Call to foo($...ARGS)
 ```
 
-### Displaying matched metavariables in rule messages
+### Anonymous metavariables
+
+Anonymous metavariables are used to specify that a metavariable exists in the pattern you want to capture.
+
+An anonymous metavariable always takes the form `$_`. Variables such as `$_1` or `$_2` are **not** anonymous. You can use more than one anonymous metavariable in a rule definition.
+
+For example, if you want to specify that a function should **always** have 3 arguments, then you can use anonymous metavariables:
+
+```yaml
+- pattern: def function($_, $_, $_)
+```
+
+An anonymous metavariable does not produce any binding to the code it matched. This means it does not store the code that it matched. The pattern:
```

Comment:
i might hesitate to say "store" in case people conflate that with the notion of Semgrep Inc storing code.

"An anonymous metavariable does not produce any binding to the code it matched. This means it does not enforce that it matches the same code at each place it is used."